File: frontend/src/types/portfolio.ts
export interface PortfolioSummary {
  total_value: number;
  cash_balance: number;
  invested_value: number;
  daily_pnl: number;
  daily_pnl_pct: number;
  total_pnl: number;
  total_pnl_pct: number;
  open_positions: number;
  winning_positions: number;
  losing_positions: number;
}

export interface Position {
  id: string;
  symbol: string;
  side: "long" | "short";
  quantity: number;
  avg_entry_price: number;
  current_price: number;
  unrealized_pnl: number;
  unrealized_pnl_pct: number;
  realized_pnl: number;
  stop_loss: number | null;
  take_profit: number | null;
  strategy_id: string | null;
  opened_at: string;
}

File: backend/app/schemas/portfolio.py
# Source: Doc 03 §4 — Pydantic portfolio şemaları
"""Portföy şemaları."""

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class PositionResponse(BaseModel):
    id: UUID
    symbol: str
    side: str
    quantity: int
    avg_entry_price: Decimal
    current_price: Decimal | None = None
    unrealized_pnl: Decimal | None = None
    unrealized_pnl_pct: Decimal | None = None
    realized_pnl: Decimal
    stop_loss: Decimal | None = None
    take_profit: Decimal | None = None
    opened_at: datetime

    model_config = {"from_attributes": True}


class PortfolioSummaryResponse(BaseModel):
    total_value: Decimal
    cash_balance: Decimal
    invested_value: Decimal
    daily_pnl: Decimal
    daily_pnl_pct: Decimal | None = None
    total_pnl: Decimal
    total_pnl_pct: Decimal | None = None
    open_positions: int
